sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_region.c

```diff
@@ -100,7 +100,7 @@ uint32_t gnrc_lorawan_pick_channel(gnrc_lorawan_t *mac)
 {
     uint8_t index = 0;
 
-    uint8_t random_number = random_uint32_range(0, bitarithm_bits_set(mac->channel_mask));
+    uint8_t random_number = random_uint32_range(0, bitarithm_bits_set(mac->channel_mask) + 1);
```

Comment:
Is for some reason `mac->channel[0]` always valid? Because when `random_number == 0` that's the chosen one.

Reply:
True. The for loop should be adapted to pick the first available channel instead.

Reply:
ah no... I think the addition should be placed outside of the function